src/pyramid_jwt/policy.py

```diff
@@ -61,12 +68,24 @@ def get_claims(self, request):
             token = request.headers.get(self.http_header)
         if not token:
             return {}
-        try:
-            claims = jwt.decode(token, self.public_key, algorithms=[self.algorithm], leeway=self.leeway)
-        except jwt.InvalidTokenError as e:
-            log.warning('Invalid JWT token from %s: %s', request.remote_addr, e)
-            return {}
-        return claims
+        audience =self.audience
+        if audience:
+            try:
+                claims = jwt.decode(token, self.public_key, algorithms=[self.algorithm],
+                                    audience=audience, leeway=self.leeway)
+                return claims
+            except jwt.InvalidTokenError as e:
+                log.warning('Invalid matched audience for JWT token from %s: %s', 
+                            request.remote_addr, e)
+                return {}
+        else:
+            try:
+                claims = jwt.decode(token, self.public_key, algorithms=[self.algorithm],
+                                    leeway=self.leeway)
+                return claims
+            except jwt.InvalidTokenError as e:
+                log.warning('Invalid JWT token from %s: %s', request.remote_addr, e)
+                return {}
```

Comment:
That code duplication looks bad. Does PyJWT do something unexpected if you pass in `audience=None` to require that? If so I suggest a different approach:

```python
extra = {}
if audience:
    extra['audience'] = audience
try:
   claims = jwt.decode(token, …, **extra)
except jwt.InvalidTokenError as e:
    …
```

Reply:
It was more to give a specific warning log that the decode failed due to an unmatched audience, without the code duplication the error is not as helpful to the user. How does this look:
```python
audience = self.audience
        def decode_jwt(self, token):
            claims = jwt.decode(token, self.public_key, algorithms=[self.algorithm],
                                    audience=self.audience, leeway=self.leeway)
            return claims

        if audience:
            try:
                decode_jwt(self, token=token)
            except jwt.InvalidTokenError as e:
                log.warning('Invalid matched audience for JWT token from %s: %s', 
                            request.remote_addr, e)
                return {}
        else:
            try:
                decode_jwt(self, token=token)
            except jwt.InvalidTokenError as e:
                log.warning('Invalid JWT token from %s: %s', request.remote_addr, e)
                return {}
```

Reply:
Why not do the `if` in the `except` just to wrap the `log.warning` call?

```python
try:
   claims = jwt.decode(token, …, audience=audience)
except jwt.InvalidTokenError as e:
    if audience:
        log.warning('Invalid matched audience')
    else:
        log.warning('Invalid JWT token')
```

Reply:
That's a good idea, I will update and test this on Monday.